### PR TITLE
store/tikv: fix context checking for coprocessor worker routines

### PR DESCRIPTION
**Pull request by hhkbp2**
Hi,

This PR tries to fix the context checking for coprocessor worker goroutines, according to the discussion in https://github.com/pingcap/tidb/pull/3010#discussion_r115902723

PTAL @tiancaiamao @zimulala @AndreMouche

### CODE REVIEW

**Comment by hanfei1991** [comment]
LGTM

File: store/tikv/coprocessor.go [hunk 1]
```diff
@@ -332,7 +332,7 @@ func (it *copIterator) work(ctx goctx.Context, taskCh <-chan *copTask) {
 		for _, resp := range resps {
 			select {
 			case ch <- resp:
-			case <-ctx.Done():
+			case <-childCtx.Done():
```
Comment (zimulala) [inline store/tikv/coprocessor.go]: Is this child canceled only by line 312?

Comment (hhkbp2) [inline store/tikv/coprocessor.go]: Yes. It's the same as `ctx.Done` semantically.

**Comment by hhkbp2** [comment]
PTAL @tiancaiamao @zimulala @shenli

**Comment by zimulala** [comment]
LGTM

**Comment by tiancaiamao** [comment]
LGTM

**Comment by hhkbp2** [comment]
PTAL @tiancaiamao

**Pull request merged**